From b3d01973b4beb8c0a2b816055207d24b68d84699 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Sat, 15 Dec 2012 21:55:50 +0800
Subject: [PORT FROM MAIN]libstagefright: enhance error handle for NuCachedSource2

BZ: 74681

enhance error handle for NuCachedSource2.When invalid offset(<0) resulted from a asf clip without
simple index object is passed to readAt,return error.

Change-Id: I400e045bb983f1fcbd169c6b4b8b925ad486fa10
Orig-Change-Id: Iff06f8e7b398119fa14a868f63e1033af370bc5f
Signed-off-by: ywan171 <yi.a.wang@intel.com>
---
 media/libstagefright/NuCachedSource2.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/NuCachedSource2.cpp b/media/libstagefright/NuCachedSource2.cpp
index 05e599b..81c9a92 100644
--- a/media/libstagefright/NuCachedSource2.cpp
+++ b/media/libstagefright/NuCachedSource2.cpp
@@ -468,6 +468,10 @@ ssize_t NuCachedSource2::readAt(off64_t offset, void *data, size_t size) {
 
     Mutex::Autolock autoLock(mLock);
 
+    if (offset < 0) {
+        return ERROR_OUT_OF_RANGE;
+    }
+
     // If the request can be completely satisfied from the cache, do so.
 
     if (offset >= mCacheOffset
-- 
1.7.4.1

